pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -399,13 +405,33 @@ func (ds *Plugin) GetLatestAttestedNodeEventID(ctx context.Context) (eventID uin
 	return eventID, nil
 }
 
+func (ds *Plugin) CreateAttestedNodeEvent(ctx context.Context, event *datastore.AttestedNodeEvent) (*datastore.AttestedNodeEvent, error) {
+	return event, ds.withWriteTx(ctx, func(tx *gorm.DB) (err error) {
+		return createAttestedNodeEvent(tx, event)
+	})
+}
+
+// DeleteAttestedNodeEvent deletes an attested node event by event ID
+func (ds *Plugin) DeleteAttestedNodeEvent(ctx context.Context, eventID uint) error {
+	return ds.withWriteTx(ctx, func(tx *gorm.DB) (err error) {
+		return deleteAttestedNodeEvent(tx, eventID)
+	})
+}
+
+// FetchAttestedNodeEvent fetches an existing event by event ID
+func (ds *Plugin) FetchAttestedNodeEvent(_ context.Context, eventID uint) (*datastore.AttestedNodeEvent, error) {
+	return fetchAttestedNodeEvent(ds.db, eventID)
```

Comment:
Each SQL command runs in a default transaction.  It is not clear how typing the code would clarify much.

If you do "SELECT * from X" that will run in an auto-transaction, so transactions are only useful when you need to ensure that you are doing two things atomically.  In this case, fetching a single row in an explicit transaction would slow the code down, as you would have to confirm the creation of a transaction for one item.

Reply:
there is an known bug, that we resolved using read transactions, that is why you can see that all our `fetch` functions are relying on that function, 
it is not a big deal here, because this function will be used only for testing, but it is always better to be consistent incase in a future we want to expose this for a real use case

Reply:
done